File: src/it/scala/com/exasol/cloudetl/kafka/docker/KafkaContainer.scala
package com.exasol.cloudetl.kafka

import org.testcontainers.containers.GenericContainer
import org.testcontainers.utility.DockerImageName

/**
 * A Kafka container that exposes a fixed port.
 *
 * This is required since [[GenericContainer]] does not provide public API for exposing fixed ports.
 */
class KafkaContainer(image: DockerImageName, fixedPort: Int = 29092) extends GenericContainer(image) {

  override def configure(): Unit = {
    super.configure()
    addFixedExposedPort(fixedPort, fixedPort)
  }

}

File: src/test/java/com/exasol/cloudetl/kafka/KafkaContainer.java
package com.exasol.cloudetl.kafka;

import org.testcontainers.containers.GenericContainer;
import org.testcontainers.utility.DockerImageName;

/**
 * A Kafka container that exposes a fixed port.
 */
class KafkaContainer extends GenericContainer<KafkaContainer> {
    private final int fixedPort;

    KafkaContainer(final DockerImageName image, final int fixedPort) {
        super(image);
        this.fixedPort = fixedPort;
    }

    @Override
    protected void configure() {
        super.configure();
        addFixedExposedPort(this.fixedPort, this.fixedPort);
    }
}
